Libplanet/Blockchain.cs

```diff
@@ -215,6 +215,49 @@ public Block<T> MineBlock(Address rewardBeneficiary)
             return block;
         }
 
+        public IEnumerable<HashDigest<SHA256>> LocateBlocks(
+            IEnumerable<HashDigest<SHA256>> locator,
```

Comment:
> In bitcoin, locator(vector of hashes) is used to find branch point. but the codes in this PR doesn't check branch yet because it's very naive implementation in this time.

I've reviewed the bitcoin implementation and found that:

1. The behavior of [FindForkInGlobalIndex()](https://github.com/bitcoin/bitcoin/blob/82ffd4d91832c275f791a17f697534cc677c89fd/src/validation.cpp#L285-L302) is not significantly different from the current implementation, except that the order is reversed and blocks other than the main chain are handled.
    - In other words, we can implement it without necessarily receiving arguments as `IEnumerable<>`.
2. The bitcoin manages the locator as `vec<uint256>`, but when it create them, [it adjusts the steps for efficient searching](https://github.com/bitcoin/bitcoin/blob/78dae8caccd82cfbfd76557f1fb7d7557c7b5edb/src/chain.cpp#L23-L49).

Though there are a number of other important concepts (e.g. block management outside the active chain), the key to this method is that the node with the hash corresponding to input returns the hashes of the blocks it should receive.

Reply:
I and @longfin verbally discussed about this, and I became to seemingly understand what the term “locator” means and satisfied with the name.  If I understood it correctly one of important things is that `IEnumerable<HashDigest<SHA256>>` is not only a set of identifiers, but a sequence of them, because it purposes to represent *path* of a blockchain.

For that purpose, `IList<HashDigest<SHA256>>` seems to describe its intention more than `IEnumerator<HashDigest<SHA256>>`.  Or it might be even better to have a distinct type to represent a locator instead.